core/classes/Group_Sync/GroupSyncManager.php

```diff
@@ -146,148 +146,6 @@ private function compileValidatorMessages(Language $language): array
         return $messages;
     }
 
-    /**
-     * Execute respective `addGroup()` or `removeGroup()` function on each of the injectors
-     * synced to the changed group.
-     *
-     * @param User   $user                   NamelessMC user to apply changes to
-     * @param string $sending_injector_class Class name of injector broadcasting this change
-     * @param array  $group_ids              Array of Group IDs native to the sending injector which were added/removed to the user
-     *
-     * @deprecated broadcastGroupChange should be used instead. This function has issues, such as that on the first sync it will replace
-     *             groups on one side with groups from the other side, depending on which side happens to sync first. On the first sync,
-     *             the desired behaviour is for the new roles on both sides to become the union of roles on both sides.
-     *
-     * @return array Array of logs of changed groups
-     */
-    public function broadcastChange(User $user, string $sending_injector_class, array $group_ids): array
```

Comment:
@Derkades Whats the main reason for removing this where we just need a set of groups vs manually setting remove/add groups?

Your NamelessLink & Plugin is only submitting changed groups to api instead of full set of groups?

However i still pretty much depend on this function as other group sync providers such as Patreon Integration i only get to set Full set of groups as i myself wont be able to detect the changed groups

This function has apperard to be working fine thought for me in all cases as i always submit the full set of groups

Reply:
Sending a list of groups works for unidirectional sync with a single authoritative source, but not for bidirectional sync. If on Discord someone has groups A, B, C and on Discord they have groups C, D, you have no idea which groups should be added or removed on which side

Reply:
I will revert this to the develop branch version and we can come back to this convo :)